### PR TITLE
Deleting Node

### PR DESCRIPTION
**Pull request by pbywater**
relates to #20 #21 

- After a few seconds of dragging a node, the menu disappears and a bin button appears
- Bin changes colour on hover
- Node deletes from the database when user drops it in the bin

### CODE REVIEW

File: src/routes/deleteMemory.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+require('env2')('./config.env');
+const { deleteMemory } = require('./../database/db_delete');
+const bodyParser = require('body-parser');
+
+const app = express.Router();
+
+app.use(bodyParser());
+
+module.exports = [
+  app.delete('/memories', (req, result) => {
```
Comment (des-des) [inline src/routes/deleteMemory.js]: this is the same as

```js
app.delete('/memories', (req, result) => { /*...*/ }

module.exports = [
  app
]
```
se #70

**Review by des-des — CHANGES_REQUESTED** [review]

File: src/server/controllers/getMemories.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const express = require('express');
+require('env2')('./config.env');
+const { getMemories } = require('./../../database/db_get');
+
+module.exports = (req, result) => {
```
Comment (des-des) [inline src/server/controllers/getMemories.js]: ok, in express we say `(req, res)` or `(request, response)`

**Comment by pbywater** [comment]
@des-des - I changed the error handling to `console.error(err)` - does that make sense?!

File: src/server/controllers/getMemories.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  getMemories(1, (err, res) => {
```
Comment (des-des) [inline src/server/controllers/getMemories.js]:
```js
module.exports = (req, res) => {
  getMemories(1, (err, memories) => {
    if(err) return res(err); // return is to exit the function, not actually return
   // same as if(err) {res(err); return; }
   result.send(memories.rows);
  }
}

File: src/server/controllers/home.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const express = require('express');
+const path = require('path');
+
+module.exports = (req, res) => {
+  res.sendFile(path.join(`${__dirname}/..` + '/public/index.html'));
```
Comment (des-des) [inline src/server/controllers/home.js]: Do not use template literal, thats what path.join does
```js
path.join(__dirname, '..', 'public', 'index.html')
```

**Review by des-des — CHANGES_REQUESTED** [review]

**Review by des-des — APPROVED** [review]

**Pull request merged**